Tensile/KernelWriterAssembly.py

```diff
@@ -12905,9 +12905,21 @@ def partialsWriteProcedure(self, kernel, vectorWidths, elements, alpha, beta, ed
     kStr += inst("s_waitcnt", "vmcnt(0)", "wait for data store")
     kStr += inst("s_barrier", "store all data before setting flag")
     kStr += inst("s_lshl_b32", sgpr(tmpSgpr), sgpr("StreamKIdx"), log2(4), "flag offset based on CTA index")
-    kStr += inst("s_mov_b32", sgpr(tmpSgpr+2), 1, "flag data")
-    kStr += inst("s_store_dword", sgpr(tmpSgpr+2), sgpr("AddressFlags", 2), sgpr(tmpSgpr), "glc", "set flag")
-    kStr += inst("s_waitcnt", "lgkmcnt(0)", "wait for flag") # TODO just for testing
+    if self.version == (9,4,2): # TODO: Temporary workaround
+      flagReg = self.vgprPool.checkOut(1, "flagReg")
```

Comment:
Is that safe to allocate vgpr here?
Can't we use tmpVgpr instead?

Reply:
I'd like to refactor more of our code to allocate temps closer to use at some point (and include names), cause passing tmps the way the code does runs a lot of risks that have caused errors in the past. But yea, this could use tmpVgpr instead, and it might reduce chances of running vgprs over the limit

Reply:
> I'd like to refactor more of our code to allocate temps closer to use at some point (and include names), cause passing tmps the way the code does runs a lot of risks that have caused errors in the past. But yea, this could use tmpVgpr instead, and it might reduce chances of running vgprs over the limit

From code design point of view, it is better to put allocation code closer to the point it is used.
However, if I remember correctly, the reason we have tmpVgpr as an argument is that allocating vgpr in this function (store code generation) might cause some conflict in vgpr allocation.
That is why I said it's safer to use tmpVgpr here.

Reply:
I did have to switch to using the tmpVgpr here to get it to pass the tests. It caused a really weird error.